feline/src/main/java/com/spotify/feline/FelineRuntime.java

```diff
@@ -72,4 +75,23 @@ public static void acceptOnExit(final Map<String, Object> data) {
       consumer.accept(data);
     }
   }
+
+  public static void addThreadLocalInitialValueConsumer(final Runnable consumer) {
+    threadLocalInitialValueConsumers.add(consumer);
+  }
+
+  public static boolean removeThreadLocalInitialValueConsumer(final Runnable consumer) {
+    return threadLocalInitialValueConsumers.remove(consumer);
+  }
+
+  public static void acceptThreadLocalInitialValue() {
+    int n = threadLocalInitialValueConsumers.size();
```

Comment:
Since this may be a frequent call, try to avoid allocation by doing the loop manually

Reply:
This might be better suited as a code comment.

Reply:
Will fix